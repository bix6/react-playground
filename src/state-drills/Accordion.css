ul {
    list-style: none;
    padding-left: 0;
    margin: 0 auto;
    text-align: center;
}

li {
    margin: 10px;
}

button {
    color: white;
    background-color: aquamarine;
    border: 2px solid indianred;
    border-radius: 25px;
    font-size: 20px;
    padding: 5px 10px;
    font-family: serif;
}

p {
    font-size: 20px;
    font-family: sans-serif;
    color: indigo;
}