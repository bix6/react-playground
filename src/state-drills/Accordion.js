import React from 'react';
import './Accordion.css';

class Accordion extends React.Component {
    static defaultProps = {sections: []};
    state = {activeSection: null};

    renderContent (index) {
        if (index === this.state.activeSection) {
            return (
                <p>
                    {this.props.sections[index].content}
                </p>
            )
        }
        return;
    }

    handleClick (index) {
        this.setState({activeSection: index})
    }

    renderList () {
        return this.props.sections.map((section, index) => (
            <li key={index}>
                <button onClick={() => this.handleClick(index)}>
                    {section.title}
                </button>
                {this.renderContent(index)}
            </li>
        ))
    }

    render () {
        return (
            <ul>
                {!!this.props.sections.length && this.renderList()}
            </ul>
        )
    }
}

export default Accordion;