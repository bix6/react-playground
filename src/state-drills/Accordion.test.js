import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Accordion Component', () => {
    // Section Data
    const sections = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
    ]

    // Smoke Test
    it('renders without errors', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Accordion />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    // Snapshots
    it('renders empty given no props', () => {
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders no sections active by default', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders a clicked section', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        wrapper.find('button').at(0).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('only renders last section when multiple clicked', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        wrapper.find('button').at(2).simulate('click')
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})