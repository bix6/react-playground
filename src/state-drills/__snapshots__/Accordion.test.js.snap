// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only renders last section when multiple clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no props 1`] = `<ul />`;

exports[`Accordion Component renders no sections active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
